storage/storageunit/storageunit.go

```diff
@@ -40,17 +42,31 @@ func NewStorageUnit(c storage.Cacher, p storage.Persister) (*Unit, error) {
 
 // NewCache creates a new cache from a cache config
 func NewCache(config CacheConfig) (storage.Cacher, error) {
-	return storageUnit.NewCache(config)
+	return factory.NewCache(config)
 }
 
 // NewDB creates a new database from database config
-func NewDB(persisterFactory storage.PersisterFactoryHandler, path string) (storage.Persister, error) {
-	return storageUnit.NewDB(persisterFactory, path)
+func NewDB(args ArgDB) (storage.Persister, error) {
+	return factory.NewDB(args)
 }
 
 // NewStorageUnitFromConf creates a new storage unit from a storage unit config
 func NewStorageUnitFromConf(cacheConf CacheConfig, dbConf DBConfig, persisterFactory storage.PersisterFactoryHandler) (*Unit, error) {
-	return storageUnit.NewStorageUnitFromConf(cacheConf, dbConf, persisterFactory)
+	if dbConf.MaxBatchSize > int(cacheConf.Capacity) {
```

Comment:
why not move the CreateWithRetries logic on storage-go, which gets as parameter the number of retries?

this way we remove this code which seems to be duplicated in storageUnit.go

Reply:
i wanted to keep NewStorageUnitFromConf as before; also since CreatWithRetries is specific to our persister implementation in mx-chain-go